### PR TITLE
New component: dxvk-gplasync

### PR DESCRIPTION
**Pull request by koplo199**
It seems dxvk-async isn't maintained anymore, a fork has been created (https://github.com/Sporif/dxvk-async/issues/64#issuecomment-1441560713) which supports the latest dxvk version in addition of allowing the use of async and gpl at the same time. Added as an unstable/experimental component.

## Type of change
- [x] New component
- [ ] Manifest fix
- [ ] Other

## Was this tested using a [Local Repository](https://maintainers.usebottles.com/Testing)?
- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by koplo199** [comment]
TODO: Just some testing

**Comment by Bananaman** [comment]
Is there any real world benefit?

Graphics pipeline library already takes care of stutters and makes it equivalent to Windows shader compilation performance for most games, by pre-compiling shader while the game's loading screens progress.

NVIDIA already has perfect performance with GPL. 

AMD has had a bad (slow and incomplete) implementation of GPL for a while, but recent drivers basically have a full implementation now.

I can't see the point of introducing a buggy library that tries to merge async with GPL. The author mentions bugs with shader caching bugs with infinitely growing cache, etc:

https://gitlab.com/Ph42oN/dxvk-gplasync/

It doesn't sound like a good library or good idea. I would be happy to learn what its real-world usefulness is, if any.

**Comment by koplo199** [comment]
> Is there any real world benefit?

Yes, basically any game that loads shaders at draw time will stutter without this patch.

I'm surprised you ask given this answer already has been given in one of the issue you personally opened (https://github.com/doitsujin/dxvk/issues/2798) and the fact that it is a well known issue even on Windows.

For the record, this exact behavior is shared by _a lot_ of UE4 games, one of if not the most used game engine in the world.
It can still affects power users who know about the `r.CreateShadersOnLoad=1` tricks in `Engine.ini`, let alone the vast majority of other users who do not.

> I can't see the point of introducing a buggy library that tries to merge async with GPL. The author mentions bugs with shader caching, bugs with infinitely growing cache, etc.

Indeed the author mentions those bugs, just before mentioning that they are... fixed.

In that regard, implying that I'm voluntary trying to introduce a known to be buggy library is at best a bit maladroit, and at worst plainly dishonest.

> It doesn't sound like a good library or good idea.

There is a good news about it: no one force you to use it. Not only `dxvx-(gpl)?async` variants are buried at the end of the list, but this one in particular is marked as `unstable` which I'm all for hiding by default (https://github.com/bottlesdevs/Bottles/issues/2825).

And whether it is or not a "good library": that's just like, your opinion, man[.](https://knowyourmeme.com/memes/thats-just-like-your-opinion-man)

> Considering that it WILL introduce new bugs, and considering that NVIDIA and AMD are now practically perfect with GPL, there needs to be a good reason to add this buggy Frankenstein patch.

You need to understand that for some people, including me, stuttering is not what we would call practically perfect. A stutter-free experience, even with some brief glitching here and there (which I personally never experienced) would, for us, be a closer fit to the definition: `dxvk-(gpl)?async` provides just that. So yes, I assure you that this is a very good reason to add this heretic piece of software (well, at least according to the DX11 conformance cult).
And actually, it would be the buggy Frankenstein patch's monster. /s

**Comment by mirkobrombin** [comment]
Can't test on my own. Anyone tested the component in Bottles? Is this ready to be merged?

**Comment by koplo199** [comment]
@mirkobrombin The latest revision only dropped 1 day ago, I tested it without issues but it's probably better to wait one week or two of testing.

I'll mark this PR as draft in the meantime, if no major issues have been reported until then I'll mark it as ready to merge again :+1:

**Comment by koplo199** [comment]
All good :ok_hand:

**Comment by koplo199** [comment]
Kind of forgot about this PR: good news is that it is now well tested.
Should be ready to merge.

File: dxvk/dxvk-gplasync-2.2-4.yml
Comment (orowith2os) [inline dxvk/dxvk-gplasync-2.2-4.yml:3]: "Maintainer" should be you, not the upstream maintainer.

Comment (koplo199) [reply dxvk/dxvk-gplasync-2.2-4.yml]: Fixed

**Review by orowith2os — APPROVED** [review]

**Pull request merged**
